### ISSUE TITLE
Default cleat location in drawer stacks/hampers should be "Below" in certain situations

### ISSUE THREAD

**Issue by TedCC1209** [CCDevNelson/Snap-Updates#287]
Let's change the default location of cleats in drawer stacks & hampers to be "Below" when placed in islands or when there is a countertop present.

**Comment by JJanneh** [comment]
@ClassyTeddy Looking at the **data_drawers.py** 
![Screenshot (148)](https://user-images.githubusercontent.com/76751383/107277711-09204f00-6a12-11eb-9137-030177a2d322.png)

**Comment by ClassyTeddy** [comment]
Hey J, this is going to be slightly similar to issue#288. You need to check upon placement (meaning inside of the `DROP_OPERATOR` class) of a drawer if the drawer's parent object is an island, or if there is a countertop part that is attached to the parent object. If either of those cases are true, then change the value of the `Cleat Location` to `"Below"`

**Comment by JJanneh** [comment]
@ClassyTeddy class OPS_Drawer_Drop(bpy.types.Operator):

**Comment by JJanneh** [comment]
@TedCC1209 @TylerStandage @ClassyTeddy Thanks for all the insight.
 **Default-Cleat-Location-Below** Ready for Testing

**Comment by TylerStandage** [comment]
@JJanneh thanks!  What branch can we test this on?

**Comment by sarahhightower** [comment]
Tested and worked as I believe it was requested. Nice job @JJanneh!

@TylerStandage I found this in the Default-Cleat-Location-Below / Nailson repository.

**Comment by ClassyTeddy** [comment]
It should also be available on the main repository

**Comment by TylerStandage** [comment]
@JJanneh I see it now, thanks!  Yes, looks like the functionality works.  One thing that we should update: 

Often the countertop is added after the drawers are already placed.  Looks like if the cleat is "above" prior to the countertop being placed, right now it remains "above" after the countertop is placed. Can we update so that when a countertop is placed, it checks to see if there are drawer stacks under it, and if so, to switch the cleat location on the drawer stack to "below"?

**Comment by JJanneh** [comment]
@TedCC1209 @TylerStandage @ClassyTeddy When a countertop is placed, drawer stacks under it, and if so, to switch the cleat location on the drawer stack to "below". Currently the drawer stacks below on placement  allowing the countertop to be placed next on top.

**Comment by TylerStandage** [comment]
@JJanneh looks good, thanks! moving to ready for merge

**Comment by ClassyTeddy** [comment]
This issue is not ready to move as it does not function with hampers like we talked about @JJanneh

**Comment by TylerStandage** [comment]
Thanks @ClassyTeddy !  @JJanneh let us know when it works for hampers as well.

**Comment by JJanneh** [comment]
@ClassyTeddy Made Changes no errors on placement but not working fully.
![Screenshot (561)](https://user-images.githubusercontent.com/76751383/110009920-ea446e00-7cda-11eb-8dfd-970c407eff81.png)
![Hampers_Drawers_](https://user-images.githubusercontent.com/76751383/110009930-efa1b880-7cda-11eb-8b58-5d8884b7acf5.png)

**Comment by ClassyTeddy** [comment]
So J, you need to get the `cleat_location` prompt from the hamper assembly just like we did it for the drawer. And I think you might be misinterpreting the goal here with the second screenshot. We don't want to change the drawer cleat location when a Hamper or Countertop is present or when placed into an island, we want to change the drawer/hamper cleat location when a countertop is present or when placed into an island. 

Basically what we did for the drawer, you need to do the exact same thing to the hamper. The drawer and the hamper should not interact with each other, they should both interact with countertops and islands. 

Let me know if that makes sense/helps

**Comment by JJanneh** [comment]
Default-Cleat-Location-Below Branch Ready for testing
@TylerStandage @TedCC1209 @ClassyTeddy Thanks once again for all the help it greatly appreciated.

**Comment by TylerStandage** [comment]
@JJanneh  I'm seeing the request working for all scenarios we've discussed.  Upon placement of countertop, it will switch cleat location to "below" if there are hampers or drawers in the section.  Also, upon placement of drawers, it will switch to "below" if there is a countertop in the section.  Islands are working as well for both hampers and drawers. @ClassyTeddy if you see the same, you can move to ready for merge.
